src/ripple/app/misc/impl/ValidatorList.cpp

```diff
@@ -187,4 +198,223 @@ ValidatorList::load (
+        return result;
+
+    boost::unique_lock<boost::shared_mutex> read_lock{mutex_};
+
+    // Update publisher's list
+    Json::Value const& newList = list["validators"];
+    publisherLists_[pubKey].sequence = list["sequence"].asUInt ();
+    std::vector<PublicKey>& publisherList = publisherLists_[pubKey].list;
+
+    std::vector<PublicKey> oldList = publisherList;
+    publisherList.clear ();
+    publisherList.reserve (newList.size ());
+    for (auto const& val : newList)
+    {
+        if (val.isObject () &&
+            val.isMember ("validation_public_key") &&
+            val["validation_public_key"].isString ())
+        {
+            auto const id = parseBase58<PublicKey>(
+                TokenType::TOKEN_NODE_PUBLIC,
+                val["validation_public_key"].asString ());
+
+            if (! id)
+                JLOG (j_.error()) <<
+                    "Invalid node identity: " <<
+                    val["validation_public_key"].asString ();
+            else
+                publisherList.push_back (*id);
+        }
+    }
+
+    // Update keyListings_ for added and removed keys
+    std::sort (
+        publisherList.begin (),
+        publisherList.end ());
+
+    auto iNew = publisherList.begin ();
+    auto iOld = oldList.begin ();
+    while (iNew != publisherList.end () ||
+        iOld != oldList.end ())
+    {
+        if (iOld == oldList.end () ||
+            (iNew != publisherList.end () &&
+            *iNew < *iOld))
+        {
+            // Increment list count for added keys
+            ++keyListings_[*iNew];
+            ++iNew;
+        }
+        else if (iNew == publisherList.end () ||
+            (iOld != oldList.end () && *iOld < *iNew))
+        {
+            // Decrement list count for removed keys
+            if (keyListings_[*iOld] == 1)
+                keyListings_.erase (*iOld);
+            else
+                --keyListings_[*iOld];
+            ++iOld;
+        }
+        else
+        {
+            ++iNew;
+            ++iOld;
+        }
+    }
+
+    if (publisherList.empty())
+    {
+        JLOG (j_.warn()) <<
+            "No validator keys included in valid list";
+    }
+
+    read_lock.unlock();
+
+    return ListDisposition::accepted;
+}
+
+ListDisposition
+ValidatorList::verify (
+    Json::Value& list,
+    PublicKey const& pubKey,
+    std::string const& blob,
+    std::string const& signature)
+{
+    boost::shared_lock<boost::shared_mutex> read_lock{mutex_};
+
+    auto iter = publisherLists_.find (pubKey);
+    if (iter == publisherLists_.end ())
+        return ListDisposition::untrusted;
+
+    auto const sig = strUnHex(signature);
+    auto const data = beast::detail::base64_decode (blob);
+    if (! sig.second ||
+        ! ripple::verify (
+            manifests_.getSigningKey (pubKey),
+            makeSlice(data),
+            makeSlice(sig.first)))
+        return ListDisposition::invalid;
+
+    Json::Reader r;
+    r.parse (data, list);
```

Comment:
We're ignoring the result from `parse` here. If it fails, we should `return ListDisposition::invalid;`